central/networkgraph/flow/datastore/internal/store/postgres/store.go

```diff
@@ -472,12 +469,26 @@ func (s *flowStoreImpl) Walk(ctx context.Context, fn func(obj *storage.NetworkFl
 func (s *flowStoreImpl) RemoveFlowsForDeployment(ctx context.Context, id string) error {
 	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.Remove, "NetworkFlow")
 
-	conn, release := s.acquireConn(ctx, ops.Remove, "NetworkFlow")
+	conn, release := s.acquireConn(ctx, ops.RemoveFlowByDeployment, "NetworkFlow")
 	defer release()
 
-	if _, err := conn.Exec(ctx, deleteDeploymentStmt, id); err != nil {
+	tx, err := conn.Begin(ctx)
+	if err != nil {
 		return err
 	}
+
+	if _, err := tx.Exec(ctx, deleteDeploymentStmt, s.clusterID, id); err != nil {
+		log.Info(err)
```

Comment:
?

Reply:
?  on the log or the fact that it I put it in a transaction?

Reply:
If we want to log the error then `log.Errorf("error deleting deployment: %v", id)`

Reply:
yeah I didn't, that was lingering from when I was having deadlock problems and missed it when I cleaned up.  I will eliminate that when I fix the merge conflict that I have to deal with.